spec/components/page_list_component/page_list_component_preview.rb

```diff
@@ -6,25 +6,27 @@ def default
   end
 
   def with_pages_and_no_conditions
-    pages = [OpenStruct.new(id: 1, question_text: "Enter your name", routing_conditions: []),
-             OpenStruct.new(id: 2, question_text: "What is your pet's phone number?", routing_conditions: []),
-             OpenStruct.new(id: 3, question_text: "How many pets do you own?", routing_conditions: [])]
+    pages = [Page.new(id: 1, position: 1, is_optional: false, question_text: "Enter your name", routing_conditions: []),
+             Page.new(id: 2, position: 2, is_optional: false, question_text: "What is your pet's phone number?", routing_conditions: []),
+             Page.new(id: 3, position: 3, is_optional: false, question_text: "How many pets do you own?", routing_conditions: [])]
```

Comment:
Why do these need to be specific model classes?

Reply:
Because we're using some of the class methods like `is_optional?` and `conditions` in the component. I guess we could mock these in an OpenStruct, but is there a reason we wouldn't want to use the model class?

Reply:
While it makes it hard to reuse these components then without the model classes. I think we should mock those out in the OpenStruct. TBH, it just means that the component requires a collection of objects and those objects should have a method called "is_optional" it shouldn't care that its a Page.